test/functional/plugin/tutor_spec.lua

```diff
@@ -0,0 +1,64 @@
+  before_each(function()
+    clear({ args = { '-u', 'NORC' } })
+    command('set cmdheight=0')
+    command('Tutor')
+  end)
+
+  it('should apply transformation', function()
+    local test_table = is_os('win') and {
+      ['!dir'] = {
+        '{2:  } 1. Type :{1:^!}{3:dir} to get a listing of your directory.                            |',
+        '{2:  }    To verify this, type :{1:^!}{3:dir} again to see your directory.                   |',
+        '{2:  } 5. Neovim will write the selected lines to the file TEST. Use :{1:^!}{3:dir} to see it|',
+        '{2:  }        :r {1:^!}{3:dir}                                                               |',
+        '{2:  }     :{1:^!}{3:dir}                   - shows a directory listing                      |',
+        '{2:  } 5. {4::r }{5:^!dir}                  reads the output of the dir command and          |',
+        '{2:  } 1. List the contents of the current directory: :{1:^!}{3:dir}                         |'
+      },
+      ['!del'] = {
+        '{2:  }        :{1:^!}{3:del} TEST                                                            |',
+        '{2:  }     :{1:^!}{3:del} FILENAME          - removes file FILENAME                          |',
+      }
+    } or {
+      ['!ls'] = {
+        '{2:  } 1. Type :{1:^!}{3:ls} to get a listing of your directory.                             |',
+        '{2:  }    To verify this, type :{1:^!}{3:ls} again to see your directory.                    |',
+        '{2:  } 5. Neovim will write the selected lines to the file TEST. Use :{1:^!}{3:ls} to see it.|',
+        '{2:  }        :r {1:^!}{3:ls}                                                                |',
+        '{2:  }     :{1:^!}{3:ls}                    - shows a directory listing                      |',
+        '{2:  } 5. {4::r }{5:^!ls}                   reads the output of the ls command and           |',
+        '{2:  } 1. List the contents of the current directory: :{1:^!}{3:ls}                          |'
+      },
+      ['!rm'] = {
+        '{2:  }        :{1:^!}{3:rm} TEST                                                             |',
+        '{2:  }     :{1:^!}{3:rm}  FILENAME          - removes file FILENAME                          |',
+      }
+    }
+
+    local screen = Screen.new(80, 1)
+    screen:set_default_attr_ids({
+      [1] = { foreground = Screen.colors.Brown, bold = true, background = Screen.colors.Yellow1 },
+      [2] = { foreground = Screen.colors.DarkBlue, background = Screen.colors.Gray },
+      [3] = { background = Screen.colors.Yellow },
+      [4] = { underline = true, foreground = tonumber('0x0088ff') },
+      [5] = { underline = true, foreground = tonumber('0x0088ff'), background = Screen.colors.Yellow },
+    })
+    screen:attach()
+
+    for search_word, search_matches in pairs(test_table) do
+      feed('gg0<CR>')
+      for _, expected_line in ipairs(search_matches) do
+        feed('/' .. search_word .. '<CR>')
```

Comment:
why is it searching for each case? isn't the transformed result visible without the cursor being on the result?

Reply:
The first instance of the replace is on line 632 and the last is on line 931, if we can have a screen that long we could do just one check, but if the contents in between those lines changes the test should be updated, even if the change is not related to the transformation itself.

Reply:
To simplify we could just test a sample of transformation, one simple and one complex, and add more if needed.

Reply:
> if we can have a screen that long we could do just one check

The screen dimensions are up to the test author:

    local screen = Screen.new(80, 100)


> but if the contents in between those lines changes the test should be updated, even if the change is not related to the transformation itself.

That's fine.